Jenkinsfile

```diff
@@ -0,0 +1,219 @@
+                                    "output": {
+                                        "to": {
+                                            "kind": "DockerImage",
+                                            "name": "${env.OPERATOR_CONTAINER_IMAGE_NAME}"
+                                        },
+                                        "pushSecret": {
+                                            "name": "quay-bot"
+                                        }
+                                    }
+                                }
+                            }
+                            """
+                            openshift.withCluster('operators-test-cluster') {
+                                openshift.withProject(env.OPENSHIFT_PROJECT_NAME) {
+                                    openshift.create(
+                                        "secret", "docker-registry", "quay-bot",
+                                        "--docker-username=${quayUsername}",
+                                        "--docker-password=${quayPassword}", "--docker-server=quay.io"
+                                        )
+                                    openshift.apply buildConfig
+                                    def build = openshift.startBuild("${env.OPERATOR_NAME}", "--from-dir=.")
+
+                                    waitUntil {
+                                        build.object().status.phase == "Running"
+                                    }
+                                    build.logs('-f')
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            post{
+                failure{
+                    echo "====++++'Build & push container image' execution failed++++===="
+                }
+            }
+        }
+        stage("Build test binary"){
+            steps{
+                dir("${env.CLONED_REPOSITORY_PATH}") {
+                    script {
+                        sh """
+                        make test/compile
+                        """
+                    }
+                }
+            }
+            post{
+                failure{
+                    echo "====++++'Build test binary' execution failed++++===="
+                    echo "Try to run 'make test/compile' locally and make sure it pass"
+                }
+            }
+        }
+        stage("Build operator-test image") {
+            steps{
+                dir("${env.CLONED_REPOSITORY_PATH}") {
+                    script {
+                        def operatorTestDockerfileContent = """
+                        FROM ${env.OPERATOR_CONTAINER_IMAGE_NAME}
+                        ADD build/_output/bin/unifiedpush-operator-test /usr/local/bin/unifiedpush-operator-test
+                        ADD deploy/operator.yaml /namespaced.yaml
+                        ADD build/test-framework/go-test.sh /go-test.sh
+                        """
```

Comment:
@grdryn I couldn't use a Dockerfile for this case since `FROM ${env.OPERATOR_CONTAINER_IMAGE_NAME}` will be always variable and I couldn't use it as an argument, since it's not supported when running binary build.

`WARNING: Specifying build arguments with binary builds is not supported`

Reply:
Hmm, shouldn't that variable already be replaced by the literal string at this point by Jenkins, before the openshift Build gets executed? Let me brush up on how strings work in Jenkins pipeline again...